src/Entity/Sylius/OrderRepository.php

```diff
@@ -245,4 +245,21 @@ public static function addVendorClause(QueryBuilder $qb, $alias, LocalBusiness $
             ->andWhere(sprintf('%s.restaurant = :restaurant', $vendorAlias))
             ->setParameter('restaurant', $restaurant);
     }
+
+    public function fetchNextSeqId(){
+        $dbConnection = $this->getEntityManager()->getConnection();
+        $nextValQuery = $dbConnection->getDatabasePlatform()->getSequenceNextValSQL('sylius_order_id_seq');
+        $id = (int) $dbConnection->executeQuery($nextValQuery)->fetchOne();
+        return $id;
+    }
+
+    public function findOneByNumber(string $number): ?OrderInterface
+    {
+        $qb = $this->createQueryBuilder('o');
+        $qb
+            ->andWhere('o.number = :number')
+            ->setParameter('number', $number);
+
+        return $qb->getQuery()->getOneOrNullResult();
+    }
```

Comment:
This shouldn't be needed to declare this method, this getter is [already exposed magically by Doctrine](https://github.com/doctrine/orm/blob/609647a51a1b28a674de270204a5cea3af57d554/lib/Doctrine/ORM/EntityRepository.php#L275-L277).

Reply:
That's magic is great! I thought that the suggestion was to use the `findOneBy` method